dhall/src/Dhall/TypeCheck.hs

```diff
@@ -2028,7 +2040,7 @@ prettyTypeMessage Untyped = ErrorMessages {..}
 
 prettyTypeMessage (InvalidPredicate expr0 expr1) = ErrorMessages {..}
   where
-    short = "Invalid predicate for ❰if❱"
+    short = "Invalid predicate for ❰if❱: "<>pretty expr1
```

Comment:
What's `expr1`? If it can be large, I'm not sure this is a good idea.

Also: Spaces around the `<>` would be nice.

Reply:
Yeah.. I ran short on time. I can possibly fix these during the week.

Is there a predefined size measure for expressions? Perhaps it could show the type only when it is small enough?

Or, perhaps show the diff against `Bool`? Somewhat more confusing, but matches the other cases?

Reply:
How about this:

```
Error: Invalid predicate for ❰if❱:
- Bool
+ Natural

1│ if 5 then 1 else 2
```

Reply:
@aleator That seems like a good idea! :+1: